### DISCUSSION TITLE
Reconfigurable intelligent surface

### DISCUSSION THREAD

**Discussion by Saman12191** [NVlabs/sionna#133] [Q&A]
Hello,

I am interested in utilizing the system models depicted below and incorporating reconfigurable intelligent surface (RIS) into them. I would appreciate your guidance on how to effectively model and simulate the RIS. Is it feasible to introduce phase shifts to the Channel Impulse Response (CIR) obtained from the system model without RIS?

<img width="523" alt="Screenshot 2023-05-19 123610" src="https://github.com/NVlabs/sionna/assets/86505056/83d9730e-2f2d-42fd-a060-ad3c5cf1f615">

<img width="527" alt="Screenshot 2023-05-19 123644" src="https://github.com/NVlabs/sionna/assets/86505056/9ae355d4-eebc-4105-933d-197b7f8e52ce">

Thanks

**Comment by jhoydis** [comment]
Hi,

Sionna RT does not provide support of RIS at the moment.

You a can add phase shifts to the channel impulse responses (these are just tensors and you can modify them in any way you want). 

We cannot give you guidance on how to model RIS, but there are numerous papers describing models that you could adopt. In general, it should not be too complicated to manually add RIS paths to those you get from the ray tracer.
